cmd/proxy/actions/app_proxy.go

```diff
@@ -98,7 +99,10 @@ func addProxyRoutes(
 	}
 	st := stash.New(mf, s, stash.WithPool(c.GoGetWorkers), withSingleFlight)
 
-	df, err := mode.NewFile(c.DownloadMode, c.DownloadURL)
+	df, err := mode.NewFile(c.DownloadMode, c.DownloadURL, errors.ConfigError(
```

Comment:
though this functional pattern is pretty awesome, it's a bit of an overkill just to add 2 lines of text to an error: 
1. We never have to customize this line
2. We can just add those two lines into the errors themselves inside `mod.NewFIle` 
3. We can still keep it functional by keeping the `mode.NewFile` signature exactly the same, but having this functional pattern internal and private to the package itself so that you don't have to manually do it a few times